adafruit_minimqtt/adafruit_minimqtt.py

```diff
@@ -535,7 +551,7 @@ def _connect(
                 rc = self._sock_exact_recv(3)
                 assert rc[0] == 0x02
                 if rc[2] != 0x00:
-                    raise MMQTTException(CONNACK_ERRORS[rc[2]])
+                    raise MMQTTException(CONNACK_ERRORS[rc[2]], code=rc[2])
```

Comment:
Looking further, would any error 0x01-0x05 be retry-able?

Reply:
@dhalbert any reason you can think of that we would retry on any of these:
```py
CONNACK_ERROR_INCORRECT_PROTOCOL: "Connection Refused - Incorrect Protocol Version",
CONNACK_ERROR_ID_REJECTED: "Connection Refused - ID Rejected",
CONNACK_ERROR_SERVER_UNAVAILABLE: "Connection Refused - Server unavailable",
CONNACK_ERROR_INCORECT_USERNAME_PASSWORD: "Connection Refused - Incorrect username/password",
CONNACK_ERROR_UNAUTHORIZED: "Connection Refused - Unauthorized",
```
The only on that I could see is a maybe would be `CONNACK_ERROR_SERVER_UNAVAILABLE`